src/lib.rs

```diff
@@ -78,69 +80,73 @@ impl Agent {
 
 // ***** Income evolution *****
 
-pub fn income(agent: Agent, rate: f32) { // add rate to global parameters
-    let a = agent.age;
-    let i = agent.social_class.init_income;
-    let g = agent.social_class.growth;
-    let s = agent.social_class.start;
-    let r = agent.social_class.retire + 1;
-    if a < s {
-        0
-    } else if a < r {
-        i * g.pow(a - s) 
-    } else {
-        i * r
+impl Agent {
+    // TODO add rate to global parameters
+    pub fn income(&self, rate: f32) -> f32 {
+        let age = self.age;
+        let init_income = self.social_class.init_income;
```

Comment:
Please no single letter variables, it feels too much like math notation ;)  Just use the full name in the code, or if it's verbose to do `self.social_class.blah`, then the intermediate variables are fine